src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -942,3 +944,435 @@ fixCopyCommand(Archive *fout, PQExpBuffer copyBuf, TableInfo *tbinfo, bool isFro
 	}
 	destroyPQExpBuffer(q);
 }
+
+/*
+ * dumpSqlvariantTableData:
+ * Dump the contents of a table with atleast one column of sql_variant data
+ * type. We use insert commands to dump the table data irrespective of value of
+ * dopt->dump_inserts. This function is derived from dumpTableData_insert in
+ * pg_dump.c modified to handle tables with sql_variant data type columns
+ * separately.
+ */
+int
+dumpSqlvariantTableData(Archive *fout, const void *dcontext)
+{
+	TableDataInfo *tdinfo = (TableDataInfo *) dcontext;
+	TableInfo  *tbinfo = tdinfo->tdtable;
+	DumpOptions *dopt = fout->dopt;
+	PQExpBuffer q = createPQExpBuffer();
+	PQExpBuffer insertStmt = NULL;
+	char	   *attgenerated;
+	bool	   *atttypnames;
+	PGresult   *res;
+	int			nfields,
+				i;
+	int			rows_per_statement = dopt->dump_inserts;
+	int			rows_this_statement = 0;
+	int 		nfields_new;
+	int 		orig_field;
+	bool 		is_sqlvariant = false;
+
+	/*
+	 * If we're going to emit INSERTs with column names, the most efficient
+	 * way to deal with generated columns is to exclude them entirely.  For
+	 * INSERTs without column names, we have to emit DEFAULT rather than the
+	 * actual column value --- but we can save a few cycles by fetching nulls
+	 * rather than the uninteresting-to-us value.
+	 */
+	attgenerated = (char *) pg_malloc(tbinfo->numatts * sizeof(char));
+	atttypnames = (bool *) pg_malloc(tbinfo->numatts * sizeof(bool));
+	atttypnames = memset(atttypnames, false, tbinfo->numatts * sizeof(bool));
```

Comment:
Should use pg_malloc0 to allocate and initialize with all false. Also, name seems to be confusing, let's name it something like `is_sqlvaraint`.

Reply:
Replaced `pg_malloc` and `memset` to `pg_malloc0`. I've changed the name of the array from atttypnames to is_sqlvariant_column.